src/wp-includes/class-wp-block-type.php

```diff
@@ -113,7 +113,7 @@ class WP_Block_Type {
 	 * Block variations.
 	 *
 	 * @since 5.8.0
-	 * @var array[]
+	 * @var array[] | callable
```

Comment:
```suggestion
	 * @var callable|array[]
```

Nit-pick

Reply:
We'll also need a `@since` tag that documents which version adds support for `callable` values.